mono/metadata/method-builder-ilgen.h

```diff
@@ -50,6 +50,8 @@ mono_mb_emit_managed_call (MonoMethodBuilder *mb, MonoMethod *method, MonoMethod
 void
 mono_mb_emit_icall (MonoMethodBuilder *mb, gpointer func);
 
+#define mono_mb_emit_icall(mb, function) mono_mb_emit_icall ((mb), (gpointer)(function))
```

Comment:
Remove that.

Reply:
It is the explicit cast vs. auto cast thing.

Reply:
Damage could be quite high here, like 100+. I'll try it.

Reply:
Maybe an inline template instead, with a cast therein? Kinda the same as the macro really, except no need to undef or put parens around the real function. Anyway I'm adding the casts, lots of them.